jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -75,9 +79,45 @@
       throws GeneralSecurityException {
     // Do not use {@link NetHttpTransport}. See {@link getConnectionFactory} for details.
     HttpTransport transport = new ApacheHttpTransport.Builder().doNotValidateCertificate().build();
+    addProxyCredentials(transport);
     return url -> new Connection(url, transport);
   }
 
+  private static void addProxyCredentials(HttpTransport transport) {
+    DefaultHttpClient httpClient =
+        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();
+
+    boolean httpProxy = System.getProperty("http.proxyHost") != null;
+    boolean httpCreds = System.getProperty("http.proxyUser") != null;
+    boolean httpsProxy = System.getProperty("https.proxyHost") != null;
+    boolean httpsCreds = System.getProperty("https.proxyUser") != null;
+    if (httpProxy && httpCreds) {
```

Comment:
Per https://github.com/GoogleContainerTools/jib/pull/1337#issuecomment-447107670, I think the logic should be

```java
    if (httpProxy && httpCreds) {
      httpClient.getCredentialsProvider().setCredentials(
          new AuthScope(http.proxyHost, http.proxyPort),
          new UsernamePasswordCredentials(http.proxyUser, http.proxyPassword));
    }
    if (httpsProxy && httpsCreds) {
      httpClient.getCredentialsProvider().setCredentials(
          new AuthScope(https.proxyHost, https.proxyPort),
          new UsernamePasswordCredentials(https.proxyUser, https.proxyPassword));
    }
```
`http.proxyHost` and `https.proxyHost` are independent, and I don't think it makes sense to set `http.proxyUser` for `https.proxyHost`. Agreed?